File: docs/examples/methodFunctions/exportVariable.ts
import { Capability, LightVM } from 'lightvm';

const vm = new LightVM({
  caps: [Capability.Control, Capability.Observe],
  runtimeConfig: { nightly: true },
});
const tools = vm.tools();

const raw = [
  ['val', 'score'],
  ['push', '2'],
  ['set', 'score'],
  ['export', 'score'],
];

const optimized = tools.optimizeBytecode(raw);
vm.load(optimized);

const scoreVar = vm.export('score');

console.log(scoreVar.call());

File: docs/examples/methodFunctions/export_variable.rs
use lightvm::LightVM;
use lightvm::types::{
  capability::Capability, runtime_config::RuntimeConfig, vmconfig::VmConfig,
};

fn main() {
  let mut vm = LightVM::new(VmConfig {
    caps: vec![Capability::Control, Capability::Observe],
    runtime_config: Some(RuntimeConfig { nightly: true }),
    ..Default::default()
  });
  let tools = vm.tools();

  let raw = r#"[
    ["val", "score"],
    ["push", "2"],
    ["set", "score"],
    ["export", "score"]
  ]"#;

  let optimized = tools.optimize_bytecode(raw);
  vm.load(optimized);

  let score_variable = vm.export("score".to_string());

  println!("{:?}", score_variable.call(&mut vm, vec![]));
}
